### PR TITLE
Feat/feature environment strategy execution reorder

### PR DESCRIPTION
**Pull request by andreas-unleash**
When reordering strategies for a feature environment:
- Adds stop when CR are enabled
- Emits an event 

## About the changes

Closes #

### CODE REVIEW

File: src/lib/services/feature-toggle-service.ts [hunk 1]
```diff
@@ -378,17 +380,92 @@ class FeatureToggleService {
...
+        const { featureName, environment, projectId: project } = context;
+        const existingOrder = (
+            await this.getStrategiesForEnvironment(
+                project,
+                featureName,
+                environment,
+            )
+        )
+            .sort((a, b) =>
+                !!a.sortOrder && !!b.sortOrder
```
Comment (sighphyre) [inline src/lib/services/feature-toggle-service.ts:423]: Can we reuse the sort done here: https://github.com/Unleash/unleash/pull/4218, it's a bit easier to read imo and while we should have strategies without sorting order, if we do, I'd rather they were handled in the same way

**Review by sighphyre — COMMENTED** [review]
This looks significant enough to want tests, or is it covered by existing tests?

File: src/lib/services/feature-toggle-service.ts (hunk 1 above)
Comment (kwasniew) [reply src/lib/services/feature-toggle-service.ts:423]: +1

File: src/lib/types/events.ts [hunk 2]
```diff
@@ -326,6 +328,43 @@ export class FeatureEnvironmentEvent extends BaseEvent {
         this.environment = p.environment;
     }
 }
+export type EnvironmentStrategyExecutionOrder = string[];
```
Comment (kwasniew) [inline src/lib/types/events.ts:331]: Can we call it StrategyIds?

File: src/lib/types/events.ts [hunk 3]
```diff
@@ -326,6 +328,43 @@ export class FeatureEnvironmentEvent extends BaseEvent {
...
+export class FeatureEnvironmentStrategyExecutionOrderUpdatedEvent extends BaseEvent {
```
Comment (kwasniew) [inline src/lib/types/events.ts:332]: I'd go for something easier to use in a conversation e.g. StrategiesOrderChangedEvent

**Review by kwasniew — CHANGES_REQUESTED** [review]
I'd write a few tests that show:
* updated order
* event emitted 
* stopped when CR enabled

File: src/test/e2e/api/admin/project/features.e2e.test.ts [hunk 4]
```diff
@@ -3227,3 +3228,130 @@ test('Enabling a feature environment should add the default strategy when only d
...
+    await addStrategies(featureName, envName);
+    const { body } = await app.request.get(
+        `/api/admin/projects/default/features/${featureName}/environments/${envName}/strategies`,
+    );
+
+    const strategies: FeatureStrategySchema[] = body;
+    let order = 1;
+    const sortOrders: SetStrategySortOrderSchema = [];
+
+    strategies.forEach((strategy) => {
```
Comment (kwasniew) [inline src/test/e2e/api/admin/project/features.e2e.test.ts:3335]: I'd replace forEach with map for readability
```
const strategies: FeatureStrategySchema[] = body;
const sortOrders: SetStrategySortOrderSchema[] = strategies.map((strategy, index) => ({
    id: strategy.id!,
    sortOrder: index + 1
}));
```

**Review by kwasniew — APPROVED** [review]
Nice. Minor comments inside.

**Pull request merged**